plugins/modules/gpg_keypair.py

```diff
@@ -0,0 +1,507 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2024, Austin Lucas Lake <53884490+austinlucaslake@users.noreply.github.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: gpg_keypair
+author: "Austin Lucas Lake (@austinlucaslake)"
+short_description: Generate or delete GPG private and public keys
+version_added: 2.20.0
+description:
+    - "This module allows one to generate or delete GPG private and public keys using GnuPG (gpg)."
+requirements:
+    - gpg >= 2.1
+attributes:
+    check_mode:
+        description:
+            - Can run in check_mode and return changed status prediction without modifying target.
+        support: full
+    diff_mode:
+        description:
+            - Will return details on what has changed (or possibly needs changing in check_mode), when in diff mode.
+        support: none
+options:
+    state:
+        description:
+            - Whether the private and public keys should exist or not, taking action if the state is different from what is stated.
+        type: str
+        default: present
+        choices: [ present, absent ]
+    key_type:
+        description:
+            - Specifies the type of key to create.
+        type: str
+        choices: [ 'RSA', 'DSA', 'ECDSA', 'EDDSA' ]
+    key_length:
```

Comment:
In the collection, the term `key size` is usually used instead of `key length`. GPG seems to use both terms (at least on the man page), so I would prefer to use `key_size` here (if you want you can add `key_length` as an alias).